source/common/tracing/http_tracer_impl.cc

```diff
@@ -40,6 +25,20 @@ static std::string valueOrDefault(const Http::HeaderEntry* header, const char* d
   return header ? header->value().c_str() : default_value;
 }
 
+static std::string buildUrl(const Http::HeaderMap& request_headers) {
+  std::string path = request_headers.EnvoyOriginalPath()
+                         ? request_headers.EnvoyOriginalPath()->value().c_str()
+                         : request_headers.Path()->value().c_str();
+  std::string url =
+      fmt::format("{}://{}{}", "http", valueOrDefault(request_headers.Host(), ""), path);
```

Comment:
Not sure what happens if http_proxy set - but the information being used in the tags is being derived from the same place as before (mainly) - i.e. the headers, so that hasn't changed.

Yes the scheme was an oversight - just updated the PR to get that from the headers.